src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -243,6 +208,50 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
         this.runByLineInProgress.set(runningByLine).ignoreErrors();
     }
 
+    private async tryToStartDebugging(mode: KernelDebugMode, editor?: NotebookEditor, cell?: NotebookCell) {
+        if (editor) {
+            if (this.notebookInProgress.has(editor.document)) {
+                return;
+            }
+
+            try {
+                this.notebookInProgress.add(editor.document);
+                if (
+                    await this.checkForIpykernel6(
+                        editor.document,
+                        mode === KernelDebugMode.RunByLine ? DataScience.startingRunByLine() : undefined
+                    )
+                ) {
+                    switch (mode) {
+                        case KernelDebugMode.Everything:
+                            this.updateToolbar(true);
+                            void this.startDebugging(editor.document);
+                            break;
+                        case KernelDebugMode.Cell:
+                            if (cell) {
+                                this.updateToolbar(true);
+                                void this.startDebuggingCell(editor.document, KernelDebugMode.Cell, cell);
+                            }
+                            break;
+                        case KernelDebugMode.RunByLine:
+                            if (cell) {
+                                this.updateToolbar(true);
+                                this.updateCellToolbar(true);
+                                await this.startDebuggingCell(editor.document, KernelDebugMode.RunByLine, cell);
```

Comment:
in line 233 we're not awaiting, but here we're awaiting.
Confused why that's the case, comments would help.

Reply:
It'll be good to await them all